src/lib/pickles/step_branch_data.ml

```diff
@@ -9,6 +9,8 @@ type ( 'a_var
      , 'a_value
      , 'ret_var
      , 'ret_value
+     , 'auxiliary_var
+     , 'auxiliary_value
      , 'max_proofs_verified
      , 'branches
      , 'prev_vars
```

Comment:
nit: Combining all this into a single new type/struct for the parameters and then passing fewer arguments could potentially improve readability and also make updates like this (where we're just adding a new field to the struct) require updating fewer places in the code.